api/plugins/tests/integration/dataplane/data_plane.go

```diff
@@ -325,8 +393,13 @@ func (dp *DataPlane) Stop() {
 	}
 
 	logger.Info("stop envoy")
-	cmd := exec.Command("docker", "stop", containerName)
-	err = cmd.Run()
+
+	if isBinaryMode() {
+		dp.cmd.Process.Signal(syscall.SIGTERM)
+	} else {
+		cmd := exec.Command("docker", "stop", containerName)
+		err = cmd.Run()
```

Comment:
I think that we can check `err` here immediately, another branch will not return an error.

Reply:
This is too trivial to change. If we add the err next time, we have to move the err check out again.